### ISSUE TITLE
image_picker is not picking up GPS in flutter

### ISSUE THREAD

**Issue by ibrahim-cse** [flutter/flutter#99817]
I am trying to make a camera app that has GPS data in its metadata. For that reason, I am using the [image_picker] package. But when I extract the metadata with [exif] package it does not show the GPS properties.

I am writing code like this:

<details>
<summary>code sample</summary>

```dart 
class MyCamera extends StatefulWidget {
  @override
  _MyCameraState createState() => _MyCameraState();
}

class _MyCameraState extends State<MyCamera> {
  String dirPath = '';
  File? imageFile;

  _initialImageView() {
    if (imageFile == null) {
      return const Text(
        'No Image Selected...',
        style: TextStyle(fontSize: 20.0),
      );
    } else {
      return Card(child: Image.file(imageFile!, width: 400.0, height: 400));
    }
  }

  _openGallery(BuildContext context) async {
    var picture = await ImagePicker().pickImage(source: ImageSource.gallery);
    setState(() {
      imageFile = File(picture!.path);
      dirPath = picture.path;
      print('path');
      print(dirPath);
    });
  }

  _openCamera(BuildContext context) async {
    var picture = await ImagePicker().pickImage(source: ImageSource.camera);
    var bytes = await picture!.readAsBytes();
    var tags = await readExifFromBytes(bytes);
    tags.forEach((key, value) => print("$key : $value"));

    setState(() {
      imageFile = File(picture.path);
      dirPath = picture.path;
      print('path');
      print(dirPath);
    });
  }

  Future<void> _showChoiceDialog(BuildContext context) {
    return showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: Text('Take Image From...'),
            content: SingleChildScrollView(
              child: ListBody(
                children: [
                  GestureDetector(
                    child: const Text('Gallery'),
                    onTap: () {
                      _openGallery(context);
                      Navigator.of(context).pop();
                    },
                  ),
                  const Padding(padding: EdgeInsets.all(8.0)),
                  GestureDetector(
                    child: const Text('Camera'),
                    onTap: () {
                      _openCamera(context);
                      Navigator.of(context).pop();
                    },
                  ),
                ],
              ),
            ),
          );
        });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('View Image'),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.spaceAround,
          children: [
            _initialImageView(),
            Column(
              children: [
                Container(
                  margin: const EdgeInsets.symmetric(horizontal: 30.0),
                  width: 250.0,
                  child: FlatButton(
                    child: const Text(
                      'Select Image',
                      style: TextStyle(color: Colors.white, fontSize: 16.0),
                    ),
                    onPressed: () {
                      _showChoiceDialog(context);
                    },
                  ),
                  decoration: BoxDecoration(
                    color: Colors.blue,
                    borderRadius: BorderRadius.circular(30.0),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }
   }
```
</details>

And these are the metadata I am getting. There is no accurate GPS data. Can you suggest what I am doing wrong?
<img src="https://user-images.githubusercontent.com/76508162/157438209-6120c593-6c86-4bc5-ace7-e3768429b29f.png"  width="300"/>

**Comment by danagbemava-nc** [comment]
Hi @ibrahim-cse, please provide the version of `image_picker` you're using as well as the output of `flutter doctor -v`. 

Thank you

**Comment by danagbemava-nc** [comment]
Hi @ibrahim-csem may I know the device that you tested on?  Judging from your screenshot, you seem to have retrieved a bit more data than I did. On my emulator and my physical device, the information I get from the images doesn't seem to match up. I'm wondering if it's an issue with image_picker or if it is just the camera not returning that data.  Kindly see logs below

<details>
<summary> exif data </summary>

### Physical device (Poco X3 NFC, Android 11)

```
I/flutter ( 3082): Image ImageWidth : 4000
I/flutter ( 3082): Image ImageLength : 3000
I/flutter ( 3082): Image Make : Xiaomi
I/flutter ( 3082): Image Model : M2007J20CG
I/flutter ( 3082): Image Orientation : Rotated 90 CW
I/flutter ( 3082): Image XResolution : 72
I/flutter ( 3082): Image YResolution : 72
I/flutter ( 3082): Image ResolutionUnit : Pixels/Inch
I/flutter ( 3082): Image DateTime : 2022:03:10 07:10:25
I/flutter ( 3082): Image YCbCrPositioning : Centered
I/flutter ( 3082): Image ExifOffset : 214
I/flutter ( 3082): Image GPSInfo : 750
I/flutter ( 3082): EXIF ExposureTime : 1/17
I/flutter ( 3082): EXIF FNumber : 189/100
I/flutter ( 3082): EXIF ExposureProgram : Program Normal
I/flutter ( 3082): EXIF ISOSpeedRatings : 2156
I/flutter ( 3082): EXIF ExifVersion : 0220
I/flutter ( 3082): EXIF DateTimeOriginal : 2022:03:10 07:10:25
I/flutter ( 3082): EXIF DateTimeDigitized : 2022:03:10 07:10:25
I/flutter ( 3082): EXIF ComponentsConfiguration : YCbCr
I/flutter ( 3082): EXIF ShutterSpeedValue : 2029/500
I/flutter ( 3082): EXIF ApertureValue : 183/100
I/flutter ( 3082): EXIF BrightnessValue : -249/50
I/flutter ( 3082): EXIF ExposureBiasValue : 0
I/flutter ( 3082): EXIF MaxApertureValue : 183/100
I/flutter ( 3082): EXIF MeteringMode : CenterWeightedAverage
I/flutter ( 3082): EXIF LightSource : D65
I/flutter ( 3082): EXIF Flash : Flash did not fire, compulsory flash mode
I/flutter ( 3082): EXIF FocalLength : 543/100
I/flutter ( 3082): EXIF SubSecTime : 851364
I/flutter ( 3082): EXIF SubSecTimeOriginal : 851364
I/flutter ( 3082): EXIF SubSecTimeDigitized : 851364
I/flutter ( 3082): EXIF FlashPixVersion : 0100
I/flutter ( 3082): EXIF ColorSpace : sRGB
I/flutter ( 3082): EXIF ExifImageWidth : 4000
I/flutter ( 3082): EXIF ExifImageLength : 3000
I/flutter ( 3082): Interoperability InteroperabilityIndex : R98
I/flutter ( 3082): Interoperability InteroperabilityVersion : [48, 49, 48, 48]
I/flutter ( 3082): EXIF InteroperabilityOffset : 719
I/flutter ( 3082): EXIF SensingMethod : Not defined
I/flutter ( 3082): EXIF SceneType : Directly Photographed
I/flutter ( 3082): EXIF ExposureMode : Auto Exposure
I/flutter ( 3082): EXIF WhiteBalance : Auto
I/flutter ( 3082): EXIF FocalLengthIn35mmFilm : 25
I/flutter ( 3082): EXIF SceneCaptureType : Standard
I/flutter ( 3082): path
I/flutter ( 3082): /data/user/0/com.example.today/cache/46fd96c0-3262-4b01-aa3b-ee60e6297860823080213435439369.jpg
```


### Emulator (Pixel 4a, Android 12)

```
I/flutter ( 4212): Image Model : sdk_gphone64_arm64
I/flutter ( 4212): Image ImageWidth : 960
I/flutter ( 4212): Image ImageLength : 1280
I/flutter ( 4212): Image DateTime : 2022:03:10 07:14:05
I/flutter ( 4212): Image Orientation : Horizontal (normal)
I/flutter ( 4212): Image ExifOffset : 144
I/flutter ( 4212): Image Make : Google
I/flutter ( 4212): EXIF FNumber : 14/5
I/flutter ( 4212): EXIF ExposureTime : 1/100
I/flutter ( 4212): EXIF SubSecTimeDigitized : 206
I/flutter ( 4212): EXIF SubSecTimeOriginal : 206
I/flutter ( 4212): EXIF SubSecTime : 206
I/flutter ( 4212): EXIF FocalLength : 5
I/flutter ( 4212): EXIF Flash : Flash did not fire
I/flutter ( 4212): EXIF ISOSpeedRatings : 100
I/flutter ( 4212): EXIF DateTimeDigitized : 2022:03:10 07:14:05
I/flutter ( 4212): EXIF DateTimeOriginal : 2022:03:10 07:14:05
I/flutter ( 4212): EXIF ExifImageLength : 1280
I/flutter ( 4212): EXIF WhiteBalance : Auto
I/flutter ( 4212): EXIF ExifImageWidth : 960
I/flutter ( 4212): EXIF ApertureValue : 297/100
I/flutter ( 4212): EXIF ShutterSpeedValue : 6643/1000
I/flutter ( 4212): EXIF ExifVersion : 0220
I/flutter ( 4212): path
I/flutter ( 4212): /data/user/0/com.example.today/cache/3a483666-646a-4f46-a25c-37b78ebabeea5920286028342825505.jpg
```
</details>

**Comment by ibrahim-cse** [comment]
Here are the data-

Physical Device (OPPO A5 2020, Android 10)

![Screenshot_2022-03-10-13-25-07-49_868a302b2e15a926b85ce26ced7d18a3](https://user-images.githubusercontent.com/76508162/157610841-ef09b241-5f03-4c4a-ae31-2e7e1b16d5d7.jpg)

Emulator (Pixel 5 API 29, Android 10)

![Screenshot_1646897131](https://user-images.githubusercontent.com/76508162/157611088-b57461f5-46b2-49a7-915b-4d5f5a7b94b1.png)

**Comment by danagbemava-nc** [comment]
Hi @ibrahim-cse, I downloaded an app that reads exif data from images, and even in the app, I cannot see any GPS information in the data that is retrieved from the image. Can you check if you're able to retrieve the GPS data? I tested a number of apps and none of them were able to retrieve any GPS data.

**Comment by ibrahim-cse** [comment]
I can not retrieve GPS data as well. But I thought image_picker automatically saves GPS data. So I was just focusing on extracting GPS data after enabling all the permission. Do you have any other way of capturing images with a camera that saves the GPS?

**Comment by danagbemava-nc** [comment]
Hi @ibrahim-cse, I did a bit of searching and it looks like the GPS data can only be saved by the image if the camera has the `save location` setting enabled and the camera has a clear view of the sky. I tested this out and your code works just fine.  I took a picture with the sky visible enough and the GPS data was recorded. See screenshot below. The image in the screenshot however was taken with my camera app and read from the gallery. Your app will probably need access to the location before you can get the data, but that is just my guess. 

<details>
<summary>screenshot</summary>

<img src="https://user-images.githubusercontent.com/88313112/157828212-4246092d-e45f-4175-bdd6-43b56256e301.jpeg" width="300" />

</details>

**Comment by ibrahim-cse** [comment]
Hi @danagbemava-nc, I tried to do this for two days but unfortunately I had no luck. Can you please tell me what package you are using in order to extract metadata? It would be a great help if you could provide me with the source code, so I can try it from my end. Thank you.

**Comment by danagbemava-nc** [comment]
Hi @ibrahim-cse, I used the code sample that you provided in your initial post along with https://pub.dev/packages/exif which is what I assumed you used. Do note that in the screenshot that I shared, the image that I used was taken with my device camera with the conditions specified in https://github.com/flutter/flutter/issues/99817#issuecomment-1064875457. I did not test with the camera option of the image picker.  Hope this helps.
